download_plugins.py

```diff
@@ -1,84 +1,124 @@
-import requests, yaml, sys
+import requests
+import yaml
+import sys
+from textwrap import dedent
+from distutils.version import LooseVersion
+
 
 # Get dependencies for a plugin, or get depedencies for a dependency.
 def get_dependencies(plugin):
-    print "Download dependencies for:", plugin
-    print "*****************************************"
-    print plugin, "has these dependencies:"
+    print(dedent("""
+    Finding dependencies for: {0}
+    ******************************************
+    {0} has these dependencies:""".format(plugin)))
 
     try:
-        dependencies = plugins_list['plugins'][plugin]['dependencies']
+        dependencies = plugins_list["plugins"][plugin]["dependencies"]
     except KeyError:
-        print "Unable to find dependencies for %s." % plugin
+        print("Unable to find dependencies for {}.".format(plugin))
+        global exit_code
         exit_code = 1
         return None
-    
+
     for dependency in dependencies:
-        print "Processing dependency: " + dependency['name']
-        
-        if (dependency['name'] not in installed_plugins) and (not dependency['optional']):
-            print "installing " + dependency['name']
+        print("Processing dependency: {}".format(dependency["name"]))
+
+        if dependency["optional"]:
+                print("{} is optional".format(dependency["name"]))
```

Comment:
It seems like there are too many indentations here? I'm surprised python didn't complain about this.